File: src/del_dup.ts
import axios from 'axios';
import {readCsv} from "./extract-cmd-only"
import * as fs from 'fs';
import csv from 'csv-parser';
import * as fastcsv from 'fast-csv';
const API_URL = 'http://localhost:5005/v1/chat/completions'; // Replace with your actual API URL
//const USER_TOKEN = ''; // Replace with your actual user token
const USER_TOKEN = '';
const inputFile = '/Users/chiwangso/Desktop/work_project/ai_cmd_score/combined.csv'; // Replace with your input file name
const input1File = '/Users/chiwangso/Desktop/work_project/ai_cmd_score/processed/res_cleared_name_pw_concated.csv'
const outputFile = '/Users/chiwangso/Desktop/work_project/ai_cmd_score/new_combined.csv'; // Replace with your input file name

function writeRowToCsv(row: string) {
    // Append the row to the CSV file
    fs.appendFile(outputFile, row + '\n', (err:any) => {
        if (err) {
            console.error('Error writing to CSV file:', err);
        } else {
            console.log('Row written to CSV file:', row);
        }
    });
}

async function main(){
    let rawData:any[] = [];
    let old_res:any = {};
    await fs.createReadStream(input1File)
        .pipe(fastcsv.parse({ })) // Adjust depending on your CSV format
        .on('data', (row: any) => {
            const concatenatedRow = `"`+row[0]+`"`;
            old_res[row[0]] = true;
            old_res[concatenatedRow] = true;
        })
        .on('end', () => {
        })
        .on('error', (error:any) => {
            console.error('Error reading CSV file:', error);
        });
    await new Promise(resolve => setTimeout(resolve, 3000));
    //console.log(old_res);
    // Read the CSV file
    await fs.createReadStream(inputFile)
        .pipe(fastcsv.parse({ })) // Adjust depending on your CSV format
        .on('data', (row: any) => {
            // Concatenate all values in the row into a single string
            const concatenatedRow = row[0];
            console.log(row)
            //console.log('Concatenated row:', concatenatedRow); // Log the concatenated result
            if(concatenatedRow in old_res){
                console.log(`${row[0]} Processed already`);
            }else rawData.push(concatenatedRow); // Store the concatenated row in rawData
        })
        .on('end', () => {

            //llm_process(rawData); // Process the concatenated data
        })
        .on('error', (error:any) => {
            console.error('Error reading CSV file:', error);
        });
        await new Promise(resolve => setTimeout(resolve, 3000));
        console.log(rawData)

}

main();

File: src/del_dup.py
import csv
import time
from pathlib import Path

API_URL = 'http://localhost:5005/v1/chat/completions'
USER_TOKEN = 'ROHqapzX0Ccdoyjcy$kbeSUekWc8tur2CN9OMhK32e9FlY1XTujm_sKkVbK8zpf8Gz1g_Bynogs10'
INPUT_FILE = '/Users/chiwangso/Desktop/work_project/ai_cmd_score/combined.csv'
INPUT1_FILE = '/Users/chiwangso/Desktop/work_project/ai_cmd_score/processed/res_cleared_name_pw_concated.csv'
OUTPUT_FILE = '/Users/chiwangso/Desktop/work_project/ai_cmd_score/new_combined.csv'

def write_row_to_csv(row):
    """Append a row to the CSV file"""
    try:
        with open(OUTPUT_FILE, 'a', newline='') as f:
            f.write(row + '\n')
        print(f'Row written to CSV file: {row}')
    except Exception as e:
        print(f'Error writing to CSV file: {e}')

def main():
    raw_data = []
    old_res = {}
    
    # Read first input file
    try:
        with open(INPUT1_FILE, 'r', newline='') as f:
            reader = csv.reader(f)
            for row in reader:
                if row:  # Check if row is not empty
                    value = row[0]
                    quoted_value = f'"{value}"'
                    old_res[value] = True
                    old_res[quoted_value] = True
                else:
                    print(f"Skipping empty row in {INPUT1_FILE}")
    except Exception as e:
        print(f'Error reading CSV file {INPUT1_FILE}: {e}')
    
    # Wait for 3 seconds
    time.sleep(3)
    
    # Read second input file
    try:
        with open(INPUT_FILE, 'r', newline='') as f:
            reader = csv.reader(f)
            for row in reader:
                if row:  # Check if row is not empty
                    value = row[0]
                    print([value])
                    if value in old_res:
                        print(f'{value} Processed already')
                    else:
                        raw_data.append(value)
                else:
                    print(f"Skipping empty row in {INPUT_FILE}")
    except Exception as e:
        print(f'Error reading CSV file {INPUT_FILE}: {e}')
    
    # Wait for 3 seconds
    time.sleep(3)
    print(raw_data)
    raw_data = list(set(raw_data))
    # Write results using csv.writer
    try:
        with open(OUTPUT_FILE, mode='w', newline='') as result_file:
            result_writer = csv.writer(result_file)
            for item in raw_data:
                result_writer.writerow([item])
    except Exception as e:
        print(f'Error writing to output CSV file: {e}')

if __name__ == '__main__':
    main()